### PR TITLE
LOP and Unoswap bugs

### PR DESCRIPTION
**Pull request by jacubcub**
just copying [PR 2117 from spellbook repo](https://github.com/duneanalytics/spellbook/pull/2117) by @fatayri to this new repository

### CODE REVIEW

**Comment by dot2dotseurat** [comment]
@jacubcub thank you! I'm adding @jeff-dude and @masquot back as the reviewers as they will be the ones to deploy this. I appreciate the flexibility.

File: deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql [hunk 1]
```diff
@@ -318,6 +318,70 @@ WITH rows AS (
...
+                select "output_returnAmount", "amount", "srcToken", "pools", "call_tx_hash", "call_trace_address", "call_block_time", "contract_address"
+                from oneinch."AggregationRouterV5_call_unoswap"
+                where call_success
+                    and call_block_time >= start_ts
+                    and call_block_time < end_ts
+
+                UNION ALL
+
+                select "output_returnAmount", "amount", "srcToken", "pools", "call_tx_hash", "call_trace_address", "call_block_time", "contract_address"
+                from oneinch."AggregationRouterV5_call_unoswap_withPermit"
```
Comment (masquot) [inline deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql]: A contract with this name does not exist in the database. Did you mean?

oneinch."AggregationRouterV5_call_unoswapToWithPermit"

Screenshot 2022-12-15 at 10 57 44

File: deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql [hunk 2]
```diff
@@ -318,6 +318,70 @@ WITH rows AS (
 
         UNION ALL
 
+        -- 1inch Unoswap for V5 Router
+        SELECT
+            call_block_time as block_time,
+            '1inch' AS project,
+            'UNI v2' AS version,
+            'Aggregator' AS category,
+            t."from" AS trader_a,
```
Comment (masquot) [inline deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql]: the traces table using alias `t` is not accessible in this outer statement. But `trader_a` is already defined on line 360, so we can probably just reference that here?

File: deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql [hunk 3]
```diff
@@ -318,6 +318,70 @@ WITH rows AS (
...
+            NULL::bytea AS trader_b,
+            "output_returnAmount" AS token_a_amount_raw,
+            "amount" AS token_b_amount_raw,
+            NULL::numeric AS usd_amount,
+            (CASE WHEN ll.to = '\xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2' AND substring("pools"[ARRAY_LENGTH("pools", 1)] from 1 for 1) IN ('\xc0', '\x40') THEN '\xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee' ELSE ll.to END) AS token_a_address,
```
Comment (masquot) [inline deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql]: runtime error: 
> Error: Missing FROM-clause entry for table "ll" at line 11, position 24.

**Review by masquot — CHANGES_REQUESTED** [review]
@fatayri - probably a few small fixes will help us bring this one over the finishing line

File: deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql (hunk 3 above)
Comment (masquot) [reply deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql]: See also here: https://dune.com/queries/1761805

**Comment by fatayri** [comment]
damn, sorry, writing this PR on a hectic day without being able to properly test didn't help much.

will fix now

**Comment by fatayri** [comment]
@masquot Fixed it here: https://dune.com/queries/1762081 

Do you want me to create a separate PR (as this one is technically not mine) or it's fine?

**Comment by masquot** [comment]
Thank you @fatayri. Let me have a look. I don't think new PR will be needed.

**Comment by masquot** [comment]
@dot2dotseurat - could you grant me rights for this repo so I can add commits to this PR directly?

@fatayri - Started writing code to DB based on the query you linked.

**Comment by bh2smith** [comment]
Looks like some of the 1inch data is "back online" although there is still a gap between december 1 and 2 days ago (1inch is PINK)

Screenshot 2022-12-15 at 14 45 48

**Comment by masquot** [comment]
> Looks like some of the 1inch data is "back online" although there is still a gap between december 1 and 2 days ago (1inch is PINK)
> 
> Screenshot 2022-12-15 at 14 45 48

Thank you @bh2smith. I wrote (partial) data to Postgres for the 15th by way of test earlier. The backfill for the full month of December has started, but is progressing extremely slowly.

Would be really great to port this code over to v2.

**Comment by bh2smith** [comment]
> Would be really great to port this code over to v2.

💯 - couldn't agree more!

**Comment by jacubcub** [comment]
> Would be really great to port this code over to v2.

Agree and something I plan on working on next couple months. Started here a while back, maybe some of it is still useful https://github.com/duneanalytics/spellbook/pull/1398

@fatayri sorry to steal your PR, just really need dex.trades back up

**Comment by bh2smith** [comment]
> just really need dex.trades back up

Ditto!

**Comment by fleupold** [comment]
> The backfill for the full month of December has started, but is progressing extremely slowly.

@masquot could you please check on the progress? It's been 24h and it seems the data hasn't caught up yet.

image

**Comment by masquot** [comment]
Thank you @fleupold. I had to run the job in daily increments to get it to finish, and, you are right, these incremental jobs ran for multiple days.

Does the data look fine again from your perspective?

**Comment by bh2smith** [comment]
Looks all good now based on this - https://dune.com/queries/328103

Screenshot 2022-12-18 at 14 59 44

**Review by masquot — APPROVED** [review]

**Pull request merged**